mlir/lib/Target/LLVMIR/Dialect/LLVMIR/LLVMToLLVMIRTranslation.cpp

```diff
@@ -341,9 +341,13 @@ convertOperationImpl(Operation &opInst, llvm::IRBuilderBase &builder,
   auto convertCall = [&](Operation &op) -> llvm::Value * {
     auto operands = moduleTranslation.lookupValues(op.getOperands());
     ArrayRef<llvm::Value *> operandsRef(operands);
-    if (auto attr = op.getAttrOfType<FlatSymbolRefAttr>("callee"))
-      return builder.CreateCall(
-          moduleTranslation.lookupFunction(attr.getValue()), operandsRef);
+    if (auto attr = op.getAttrOfType<FlatSymbolRefAttr>("callee")) {
+      llvm::Function *callee =
+          moduleTranslation.lookupFunction(attr.getValue());
+      llvm::CallInst *call = builder.CreateCall(callee, operandsRef);
+      call->setCallingConv(callee->getCallingConv());
```

Comment:
Are there any calling conventions (i.e., not SPIR-V) in LLVM that might require a different CC on the function (callee) and the actual call instruction? 

If such thing exists (I'm not aware of any though), this might break.

Reply:
From [LLVM doc](https://llvm.org/docs/LangRef.html#calling-conventions):
```quote
The calling convention of any pair of dynamic caller/callee must match, 
or the behavior of the program is undefined.
```